packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -393,21 +389,29 @@ library LibProving {
             }
         } else {
             if (_sameTransition) revert L1_ALREADY_PROVED();
+
             // Contest the existing transition and prove it to be invalid. The new prover get all
             // rewards.
             reward = _rewardAfterFriction(_ts.validityBond);
-            _ts.contestations += 1;
+
+            uint96 livenessBond = _blk.livenessBond;
+            if (livenessBond != 0) {
+                if (_blk.assignedProver == msg.sender) {
```

Comment:
Could you explain this if statement? I don't think I follow. I'm not sure why or how the assigned prover would be `msg.sender` here (also together with some limitations on assignedProver in `_checkProverPermission`).

Reply:
I guess you asked the question mainly due to the incorrect comment in this section (updated).

When the block is proposed, `blk.livenessBond` is nonzero; if the assigned prover managed to prove the block in the proving window, it also enters this `_overrideWithHigherProof`'s second if-statement, then `msg.sender (the prover) == blk.assignedProver` will be true; if another prover proves the block out of the proving window, the if-statement will be false.